pkg/types/openstack/validation/platform.go

```diff
@@ -34,6 +37,13 @@ func ValidatePlatform(p *openstack.Platform, n *types.Networking, fldPath *field
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("ingressVIP"), p.IngressVIP, err.Error()))
 	}
 
+	// Ensure clusterNetwork doesn't overlap with any machine network
+	for _, cn := range n.ClusterNetwork {
+		if err := validateNoOverlapMachineCIDR(&cn.CIDR.IPNet, n); err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("networking").Child("clusterNetwork"), cn.CIDR.IPNet.String(), err.Error()))
+		}
+	}
```

Comment:
how is this any different from https://github.com/openshift/installer/blob/d91c40bca7952bd89730991f949cb10d16a71f9b/pkg/types/validation/installconfig.go#L334-L338